### DISCUSSION TITLE
Clearing State/Form once submitted

### DISCUSSION THREAD

**Discussion by Biz-e-Services** [react-hook-form/react-hook-form#3070] [Q&A] [answered]
Hi. I'm using Little-State-Machine (Form Wizard) and was wondering how to clear the state/form once submitted?
Tried using reset({}) on my last page in the OnSubmit event, but it does not clear. When I return to the form it populates with previous state data. Any help appreciated.

```
const onSubmit = async (requestDetail) => {
        await auth.validateSignature(requestDetail.requestSignAuthorisationId, requestDetail.authorisationPassword);
        const detail = { ...requestDetail };
        delete detail.authorisationPassword;
        action(detail);
         await PermitDataService.savePermit(state);
         reset({???}) <---- Need to clear State in some way once submitted from last page in wizard (Little State Machine)
         props.history.push("/PermitREQList");
    };
```

**Comment by bluebill1049** [comment, marked as answer]
because little state machine is persisted your form data in session storage, if you don't need to persist data, just use react context and you don't need this package.

**Reply by Biz-e-Services** [reply]
Thanks Bill. I thought that because I was using the Form Wizard  across multiple steps/pages and needed to persist the data across all forms? Is this not so using https://codesandbox.io/s/react-hook-form-wizard-form-gxvvc

**Reply by Biz-e-Services** [reply]
Nuts, I just did a read up on your suggestion to use react context. A bit over my head at mo. They also mention component composition. Which, in your opinion, is better in light of a "Wizard" and can you point me to any code examples that use it in a wizard?
